app/code/core/Mage/Adminhtml/Controller/Action.php

```diff
@@ -151,12 +151,12 @@ public function preDispatch()
     {
         // override admin store design settings via stores section
 
-        // get theme form backend config
         $legacy = Mage::getStoreConfig('admin/design/theme');
```

Comment:
Not needed anymore ;)

Reply:
followed your Suggestion now! Thinking about it today, it's the right way.